x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -377,9 +377,9 @@ private LogicalPlan resolveAggregate(Aggregate a, List<Attribute> childrenOutput
                 }
             }
 
-            if (a.expressionsResolved() == false && Resolvables.resolved(groupings)) {
+            if (a.expressionsResolved() == false) {
                 AttributeMap<Expression> resolved = new AttributeMap<>();
-                for (Expression e : groupings) {
+                for (Expression e : groupings.stream().filter(Expression::resolved).toList()) {
```

Comment:
Use a `for` loop over the groupings with a simple `if` for `Expression::resolved` (instead of the streaming approach), it's a less complex structure.